# HG changeset patch
# User Miro Hrončok <miro@hroncok.cz>
# Date 1528476582 -7200
# Branch py37
# Node ID c5a2c7e3191cdcebf3fa5f0e52f55910ed2120b9
# Parent  0e5a48796ab969d874c6b772c5c33561ac2d1b0d
Don't raise StopIteration from generator, return instead

See https://www.python.org/dev/peps/pep-0479/

diff --git a/paste/auth/digest.py b/paste/auth/digest.py
--- a/paste/auth/digest.py
+++ b/paste/auth/digest.py
@@ -57,7 +57,7 @@
         prev = item
 
     yield prev.strip()
-    raise StopIteration
+    return
 
 def _auth_to_kv_pairs(auth_string):
     """ split a digest auth string into key, value pairs """
